WordPress/Classes/Utility/Media/MediaLoader.swift

```diff
@@ -0,0 +1,89 @@
+
+/// Protocol used to abstract the information needed to load post related images
+///
+@objc protocol PostInformation {
+
+    /// The post is privated, and hosted in WPcom.
+    /// Redundant name due to naming conflict.
+    ///
+    var isPostPrivate: Bool { get }
+
+    /// The blog is self-hosted and there is already a basic auth credential stored.
+    ///
+    var isBlogSelfHostedWithCredentials: Bool { get }
+}
+
+@objc class MediaLoader: NSObject {
```

Comment:
This isn't really a media loader, since it only deals with images. In that regard, I think `ImageLoader` would be a improvement. Also, since this is only used in `PostCardTableViewCell`, it's probably too generic. `PostCardImageLoader` is even better, imo.

Reply:
The idea is to extend it a bit more as necessary and use this class in as many places as possible.
`+1` to change the name to `ImageLoader`. 🙂